packages/jsii-pacmak/lib/targets/go/types/class.ts

```diff
@@ -18,9 +18,10 @@ export class GoClass extends GoStruct {
   public constructor(pkg: Package, public type: ClassType) {
     super(pkg, type);
 
-    this.methods = Object.values(this.type.getMethods(true)).map(
-      (method) => new ClassMethod(this, method),
-    );
+    this.methods = Object.values(this.type.getMethods(true))
```

Comment:
Instead of emitting at the package level, can we just do
```suggestion
    this.methods = Object.values(this.type.getMethods(true)).map(
      (method) => method.static ? new StaticMethod(this, method) : new ClassMethod(this, method)
      )
```
And emit normally as part of the class? Since static methods are owned and defined by classes I don't see the need for the package to be aware of them in particular.

Since `StaticMethod` inherits from `ClassMethod`, no change in types should be necessary.

Reply:
As discussed previously with @jasdel, we decided that we were going to emit at package level. See: https://github.com/SoManyHs/aws-cdk-rfcs/blob/master/text/204-golang-bindings.md#case-1-simple-class

Reply:
Having their emit logic handled by the the `GoClass` doesn't change where they are emitted. They are still at the top level of the package where the class is defined similar to the pointer receivers.